generator/lib/src/code_chunks.dart

```diff
@@ -374,58 +374,61 @@ class CodeChunks {
       return readField();
     }).toList(growable: false);
 
-    // add initializers for relations
-    entity.properties.forEachIndexed((int index, ModelProperty p) {
-      if (p.isRelation) {
-        postLines.add(
-            'object.${propertyFieldName(p)}.targetId = ${fieldReaders[index]};'
-            '\n object.${propertyFieldName(p)}.attach(store);');
-      }
-    });
-
-    postLines.addAll(entity.relations.map((ModelRelation rel) =>
-        'InternalToManyAccess.setRelInfo(object.${rel.name}, store, ${relInfo(entity, rel)}, store.box<${entity.name}>());'));
-
-    postLines.addAll(entity.backlinks.map((ModelBacklink bl) {
-      return 'InternalToManyAccess.setRelInfo(object.${bl.name}, store, ${backlinkRelInfo(entity, bl)}, store.box<${entity.name}>());';
-    }));
-
     // try to initialize as much as possible using the constructor
     entity.constructorParams.forEachWhile((String declaration) {
       // See [EntityResolver.constructorParams()] for the format.
-      final paramName = declaration.split(' ')[0];
-      final paramType = declaration.split(' ')[1];
+      final declarationParts = declaration.split(' ');
+      final paramName = declarationParts[0];
+      final paramType = declarationParts[1];
+      final paramDartType = declarationParts[2];
 
       final index = fieldIndexes[paramName];
-      if (index == null) {
+      late String paramValueCode;
+      if (index != null) {
+        paramValueCode = fieldReaders[index];
+      } else if (paramDartType.startsWith('ToMany<')) {
+        paramValueCode = 'ToMany()';
+      } else {
         // If we can't find a positional param, we can't use the constructor at all.
-        if (paramType == 'positional') {
-          log.warning("Cannot use the default constructor of '${entity.name}': "
+        if (paramType == 'positional' || paramType == 'required-named') {
+          throw InvalidGenerationSourceError(
+              "Cannot use the default constructor of '${entity.name}': "
               "don't know how to initialize param $paramName - no such property.");
-          constructorLines.clear();
-          return false;
         } else if (paramType == 'optional') {
           // OK, close the constructor, the rest will be initialized separately.
           return false;
         }
         return true; // continue to the next param
       }
 
+      if (index != null && entity.properties[index].isRelation) {
```

Comment:
Can this be part of the `index != null` block above that also assigns `paramValueCode`?

Reply:
moved